### ISSUE TITLE
Bug report: AttnBlock3D reshape disorder

### ISSUE THREAD

**Issue by SamitHuang** [PKU-YuanGroup/Open-Sora-Plan#171]
There is a bug when reshaping the spatial-temporal input for computing spatial attention, [here](https://github.com/PKU-YuanGroup/Open-Sora-Plan/blob/main/opensora/models/ae/videobase/modules/attention.py#L59)

``` python
        b, c, t, h, w = q.shape
        q = q.reshape(b * t, c, h * w)
```

Correct one is:
```python
        b, c, t, h, w = q.shape
        q = q.permute(0, 2, 1, 3, 4)  # (b t c h w)
        q = q.reshape(b * t, c, h * w)
```
Without transposing T-axis , the values may not be grouped correctly in reshaping.

Another reference is  the MagViT-v2 torch impl. 
https://github.com/lucidrains/magvit2-pytorch/blob/main/magvit2_pytorch/magvit2_pytorch.py#L443-L444

**Comment by LinB203** [comment]
Sorry for that. We merge that to fix this bug.

**Comment by SamitHuang** [comment]
> Sorry for that. We merge that to fix this bug.

thanks. btw, since the computation logic is changed, the model may require re-training.
